### PR TITLE
Add models to do batched updates

### PR DESCRIPTION
**Pull request by keis**
This allows setting multiple properties in one HTTP request to the
nanoleaf device. There's no noticeable difference but it avoids some
issue in theory if one request was lost or delayed (due to a
network blip, process not getting scheduled for some time on low end
hardware, et c) which otherwise would only apply one of the wanted
updates to be applied.

Adds a few utility methods to help define these models easily.

### CODE REVIEW

**Comment by keis** [comment]
As discussed in #1 I think this would be a safer and faster way of doing multiple property changes.

Not sure if the meta magic is something you want in the project but I find removing these repeated patterns makes for nicer code. Let me know if it's not clear how some of it works.

**Comment by Oro** [comment]
There is a bit of meta magic going on here, but if this reduces boilerplate I'm all for it, nice going!

The only thing that is not obvious to me is the `deepsetter`, could you add a comment there why the slice is necessary?  

Also (but that might just be because I don't fully understand Python and is more a nitpick than anything), could you please prefix the internal methods with `_`? It is my understanding that they signal internal use only.

**Comment by keis** [comment]
sure thing, added

the idea with the slice in the deepsetter is so that when trying set e.g `foo.bar.baz` to `value` the loop will create `{'foo': {'bar': {}}` and leaving data as the empty dict identified by `bar`. The final step is then setting `data['baz'] = 'value'` and we get the desired result of `{'foo': {'bar': {'baz': 'value'}}`. Without the slice we would instead get something like `{'foo': {'bar': {'baz': {'??': 'value'}}}`

File: pynanoleaf/pynanoleaf.py [hunk 1]
```diff
@@ -4,8 +4,74 @@
...
+        super().__init__(data or {})
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+
 class Nanoleaf(object):
     """Nanoleaf API wrapper"""
+
+    Info = Info
+    State = State
```
Comment (Oro) [inline pynanoleaf/pynanoleaf.py]: Sorry, should've asked that before: Do you want to expose `State` and `Info` to the outside? I'd say they should also be internalized here. (Not the Classes, just the references inside the Nanoleaf object itself)

Comment (keis) [reply pynanoleaf/pynanoleaf.py]: Yeah, or at least State, having this as a "factory method" of sorts on the instance is a nice convenience, both could also be used for type annotations. I was thinking of `pynanoleaf.Nanoleaf.State` (or someinstance.State) as the proper name for these that should be used in code using the module. So I would rather change the module level classes to be `_Info` and `_State`

Comment (Oro) [reply pynanoleaf/pynanoleaf.py]: Works for me.  
Please do change the the module wide classes to internal then and I'll cut a new release. And thanks again for the PRs!

Comment (keis) [reply pynanoleaf/pynanoleaf.py]: Done, and thanks for creating the module in the first place :hand:

**Pull request merged**